portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -126,8 +143,18 @@ private void postComment(HttpServletRequest request, HttpServletResponse respons
     String text = getParameter(request, "text-input", "");
     Date date = new Date();
 
+    // Get the URL of the image that the user uploaded to Blobstore.
+    // If no image was uploaded, this will simply be undefined and will be 
+    // handled on the frontend. If a non-image was uploaded, redirect 
+    // back to HTML page.
+    String imageUrl = getUploadedFileUrl(request, "image");
+    if (imageUrl != null && imageUrl.equals(NOT_AN_IMAGE_EXCEPTION)) {
```

Comment:
did you mean to have an '||' here instead?

Reply:
Yes, thank you!

Reply:
Actually, I take this back -- I meant to have a `||` here. If the `imageUrl` is `null`, then that means the user only posted a comment, in which case I want to post the comment with no image. Otherwise, I want to check that the `imageUrl` is actually an image, but if I do this without verifying that the `imageUrl != null`, then I could get a `NullPointerException`.

Reply:
If the `imageUrl` is `null`, that is an expected behavior, and setting this variable to `null` within the Comment and Entity is handled.